src/codegen.cpp

```diff
@@ -7117,18 +7102,16 @@ jl_compile_result_t jl_emit_codeinst(
         // Prepare debug info to receive this function
         // record that this function name came from this linfo,
         // so we can build a reverse mapping for debug-info.
-        if (!JL_HOOK_TEST(params.params, module_activation)) {
-            bool toplevel = !jl_is_method(codeinst->def->def.method);
-            if (!toplevel) {
-                const DataLayout &DL = m->getDataLayout();
-                // but don't remember toplevel thunks because
-                // they may not be rooted in the gc for the life of the program,
-                // and the runtime doesn't notify us when the code becomes unreachable :(
-                if (!specf.empty())
-                    jl_add_code_in_flight(specf, codeinst, DL);
-                if (!f.empty() && f != "jl_fptr_args" && f != "jl_fptr_sparam")
-                    jl_add_code_in_flight(f, codeinst, DL);
-            }
+        bool toplevel = !jl_is_method(codeinst->def->def.method);
+        if (!toplevel) {
+            const DataLayout &DL = m->getDataLayout();
+            // but don't remember toplevel thunks because
+            // they may not be rooted in the gc for the life of the program,
+            // and the runtime doesn't notify us when the code becomes unreachable :(
+            if (!specf.empty())
+                jl_add_code_in_flight(specf, codeinst, DL);
+            if (!f.empty() && f != "jl_fptr_args" && f != "jl_fptr_sparam")
+                jl_add_code_in_flight(f, codeinst, DL);
         }
```

Comment:
I think this code can move into jitlayers.cpp now?

Reply:
You mean the entirety of `jl_emit_codeinst`, or just this snippet? I'll merge this PR first, but we can do the NFC changes afterwards.